codegen/test_data/clients/bar.gogen

```diff
@@ -216,14 +303,38 @@ func (c *BarClient) TooManyArgs(
 		respHeaders[k] = res.Header.Get(k)
 	}
 
-	res.CheckOKResponse(200)
-
-	var responseBody clientsBarBar.BarResponse
-	err = res.ReadAndUnmarshalBody(&responseBody)
-	if err != nil {
-		return nil, respHeaders, err
+	res.CheckOKResponse([]int{200, 403})
+
+	switch res.StatusCode {
+	case 200:
+		var responseBody clientsBarBar.BarResponse
+		err = res.ReadAndUnmarshalBody(&responseBody)
+		if err != nil {
+			return nil, respHeaders, err
+		}
+
+		return &responseBody, respHeaders, nil
+
+	case 403:
+		var exception clientsBarBar.BarException
+		err = res.ReadAndUnmarshalBody(&exception)
+		if err != nil {
+			return nil, respHeaders, err
+		}
+		return nil, respHeaders, &exception
+
+	default:
```

Comment:
how about removing this from default and then you don't need to worry about the dead code check below? Then anyone adding a status code but forgetting to return will see "Unexpected http client response" errors when testing/building.